### ISSUE TITLE
Internal error in Agda/TypeChecking/Monad/Context.hs:120

### ISSUE THREAD

**Issue by bafain** [agda/agda#4375]
Agda version: `2.6.1-ef74b1d`
Agda standard library version: `1.2-6ce46df6`

Code:
```
module Error (_ : Set) where

open import Agda.Builtin.Unit  
-- open import Data.Unit

_ = Set
```

Steps to reproduce:
1. Open the code in `emacs` and load the file.
2. Uncomment the import statement `open import Data.Unit` and reload the file.

**Comment by nad** [comment]
I could reproduce the problem, and I could fix it by removing all interface files.

**Comment by bafain** [comment]
These steps produce the same error:
1. Remove all interface files: `find $AGDAPRIM $AGDASTDLIB Error.agdai -name '*.agdai' -delete`
2. Open the code in `emacs` and load the file.
3. Uncomment the import statement `open import Data.Unit` and reload the file.
4. Comment the import statement again and reload the file.

**Comment by sattlerc** [comment]
Here is a smaller, self-contained test case:
```agda
module Issue4375.A where
```
```agda
module Issue4375.B where

primitive
  primForce : _
```
```agda
module Issue4375 where

module _ (S : Set) where
  import Issue4375.A
  import Issue4375.B

  X : Set
  X = S
```
Remove interface files, open Agda in interaction mode, check Issue4375, comment out the line `import Issue4375.B`, then check the file again. Result:
```
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/Monad/Context.hs:120
```

**Comment by sattlerc** [comment]
The issue disappears with option `--no-caching`, so it is caching related.

**Comment by UlfNorell** [comment]
The problem was that rebinding primitives from imported modules (happens during scope checking), changed the next fresh checkpoint id (in the TCState). Caching then loaded a cached state with checkpoint ids not matching the current ones. Fix in #4385 prevents computing primitives from updating the state.
